ejb/WFLY-13354-galleon-for-ejb.adoc

```diff
@@ -0,0 +1,110 @@
+= Galleon layers for the EJB subsystem
+:author:            Brian Stansberry; Yeray Borges; Cheng Fang
+:email:             cfang@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+The goal of this feature is to supply a list of https://docs.wildfly.org/galleon/#_layers[Galleon layers]
+for the EJB subsystem to support common predefined EJB feature sets.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-13354[WFLY-13354 Add Galleon layers for the EJB subsystem]
+* https://issues.redhat.com/browse/EAP7-1444[EAP7-1444 Galleon Layers in EAP XP 2]
+
+=== Related Issues
+* https://issues.redhat.com/browse/WFLY-13433[Improve capability support in EJB3 subsystem]
+* https://issues.redhat.com/browse/WFLY-13356[Add a Galleon layer for Webservices]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+
+=== Testing By
+
+* [X] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+WildFly
+
+=== Other Interested Projects
+
+== Requirements
+
+These analysis and requirements are based on https://github.com/bstansberry/wildfly/commits/ejb-layers2[Brian's proposal].
+
+If a user is trying to slim their server I think of four key things they want to do in descending order of importance:
+
+1. Eliminate unnecessary open ports.
+
+2. Eliminate unnecessary exposure.
+
+3. Eliminate jars, both to save filesystem footprint and to reduce any theoretical attack surface.
+
+4. Reduce configuration clutter and unnecessary services.
+
+Based on the above factors, the following layers are to be supported:
+
+1. ejb3-lite. This is like what we provide in standalone.xml, but without the MDB instance pool or the remote connector resource.
```

Comment:
Do we need the 3 in all the layer names?  This is the major version number of the specification but we don't currently plan to support any other ejb version in parallel do we?

The reason I ask is it came up late in the development of JBoss AS7 that really the subsystem should not have contained the 3 but it was too late at that point.

Reply:
Good point. On the other hand, I think ejb3 distinghish itself from old-style, heavy-weight ejb 2.x, so it conveys some positive connotation. I'm neutral on this naming, but if there is strong perference to drop the 3, I'm fine with renaming it.